src/bacnet/basic/object/netport.c

```diff
@@ -1840,113 +1840,36 @@ bool Network_Port_BBMD_FD_Table_Set(uint32_t object_instance, void *fdt_head)
- * @param  a - ip-address first octet
- * @param  b - ip-address first octet
- * @param  c - ip-address first octet
- * @param  d - ip-address first octet
+ * @param  bbmd_address - HostNPort FD BBMD Address
  *
- * @return  true if ip-address was retrieved
+ * @return  true if BBMD Address was retrieved
  */
-bool Network_Port_Remote_BBMD_IP_Address(
-    uint32_t object_instance, uint8_t *a, uint8_t *b, uint8_t *c, uint8_t *d)
+bool Network_Port_Remote_BBMD_Address(
+    uint32_t object_instance, BACNET_HOST_N_PORT *bbmd_address)
 {
-    unsigned index = 0; /* offset from instance lookup */
     bool status = false;
+    const unsigned index = Network_Port_Instance_To_Index(
+        object_instance); /* offset from instance lookup */
 
-    index = Network_Port_Instance_To_Index(object_instance);
     if (index < BACNET_NETWORK_PORTS_MAX) {
+        BACNET_HOST_N_PORT *BBMD_Address = NULL;
+
         if (Object_List[index].Network_Type == PORT_TYPE_BIP) {
-            if (a) {
-                *a = Object_List[index].Network.IPv4.BBMD_IP_Address[0];
-            }
-            if (b) {
-                *b = Object_List[index].Network.IPv4.BBMD_IP_Address[1];
-            }
-            if (c) {
-                *c = Object_List[index].Network.IPv4.BBMD_IP_Address[2];
-            }
-            if (d) {
-                *d = Object_List[index].Network.IPv4.BBMD_IP_Address[3];
-            }
-            status = true;
+            BBMD_Address = &Object_List[index].Network.IPv4.BBMD_Address;
+        } else if (Object_List[index].Network_Type == PORT_TYPE_BIP6) {
+            BBMD_Address = &Object_List[index].Network.IPv6.BBMD_Address;
         }
-    }
-
-    return status;
-}
 
-/**
- * For a given object instance-number, sets the ip-address
- * Note: depends on Network_Type being set for this object
- *
- * @param  object_instance - object-instance number of the object
- * @param  a - ip-address first octet
- * @param  b - ip-address first octet
- * @param  c - ip-address first octet
- * @param  d - ip-address first octet
- *
- * @return  true if ip-address was set
- */
-bool Network_Port_Remote_BBMD_IP_Address_Set(
-    uint32_t object_instance, uint8_t a, uint8_t b, uint8_t c, uint8_t d)
-{
-    unsigned index = 0; /* offset from instance lookup */
-    bool status = false;
-
-    index = Network_Port_Instance_To_Index(object_instance);
-    if (index < BACNET_NETWORK_PORTS_MAX) {
-        if (Object_List[index].Network_Type == PORT_TYPE_BIP) {
-            if ((Object_List[index].Network.IPv4.BBMD_IP_Address[0] != a) ||
-                (Object_List[index].Network.IPv4.BBMD_IP_Address[1] != b) ||
-                (Object_List[index].Network.IPv4.BBMD_IP_Address[2] != c) ||
-                (Object_List[index].Network.IPv4.BBMD_IP_Address[3] != d)) {
-                Object_List[index].Changes_Pending = true;
+        if (BBMD_Address) {
+            if (bbmd_address) {
+                *bbmd_address = *BBMD_Address;
```

Comment:
I prefer to use a function to copy structures.  See host_n_port_copy().

Reply:
done.